pandas/core/construction.py

```diff
@@ -795,6 +795,7 @@ def _try_cast(
             shape = arr.shape
             if arr.ndim > 1:
                 arr = arr.ravel()
+            arr = ensure_wrapped_if_datetimelike(arr)
```

Comment:
does the test added in this PR address the comment on L792?

Reply:
ensure_string_array treats EAs specially, and in this case both checks on lines 748 and 750 would pass, and the array would get converted to object by two consecutive `astype` calls: first to string and finally to object, which works as expected.

However, if we do not cast it to an EA, the check on line 748 (in ensure_string_array) will fail, which leads to the array being cast directly to object dtype by `np.asarray` on line 760:

https://github.com/pandas-dev/pandas/blob/bb0fcc23eed9f6a1a6506c6e27b98fb397ce747e/pandas/_libs/lib.pyx#L748-L760

This is especially problematic for `M8[ns]` arrays since numpy converts the dates into nanosecond epochs (strangely this doesn't seem to happen with any other M8 dtype array, just the nanosecond one).

The resulting Series will then contain these epochs, instead of the usual YYYY-MM-DD HH:MM:SS format, which cannot be easily converted back to dt64.

I thought casting it to DatetimeArray was a good way of preventing this from happening, but you know better than me so I'm open to suggestions.

Reply:
> does the test added in this PR address the comment on L792?

I guess it does, would you like me to remove it?

Reply:
@jbrockmendel Hi, sorry to be a bother, but would it be possible to get some feedback?
Thanks!

Reply:
cc @WillAyd @mroeschke

Reply:
Where does the nanosecond behavior diverge from other units for M8 arrays? I see you hinting at that but I'm not sure if it has been identified already.

I do agree it is strange for only nanoseconds to diverge from the other units

Reply:
Yeah so whenever we cast an M8 nparray to string using the Series constructor we get a Series with object dtype containing stringified dates. We can recover the original timestamps from this Series easily.

Here is the described behaviour using a M8[ms] array (there is no particular reason for picking milliseconds since this next behaviour is observed with any precision **other than nanoseconds** from what I have seen):
```
>>> arr = np.array([
...     ('2024-03-17T00:00:00.000000000'),
...     ('2024-03-18T00:00:00.000000000'),
... ], dtype="M8[ms]")
>>> arr
array(['2024-03-17T00:00:00.000', '2024-03-18T00:00:00.000'],
      dtype='datetime64[ms]')
>>> s1 = pd.Series(arr, dtype=str)
>>> s1
0    2024-03-17 00:00:00
1    2024-03-18 00:00:00
dtype: object
>>> s2 = pd.Series(s1, dtype='M8[ns]')
>>> s2
0   2024-03-17
1   2024-03-18
dtype: datetime64[ns]
```
However, if (and only if) the nparray has nanosecond precision, we get a Series with object dtype containing unix timestamps instead of the usual stringified date format. The original timestamps cannot be easily recovered from the resulting Series:
```
>>> arr = np.array([
...     ('2024-03-17T00:00:00.000000000'),
...     ('2024-03-18T00:00:00.000000000'),
... ], dtype="M8[ns]")
>>> arr
array(['2024-03-17T00:00:00.000000000', '2024-03-18T00:00:00.000000000'],
      dtype='datetime64[ns]')
>>> s1 = pd.Series(arr, dtype=str)
>>> s1
0    1710633600000000000
1    1710720000000000000
dtype: object
>>> s2 = pd.Series(s1, dtype='M8[ns]')
>>> s2
0    1710633600000000000
1    1710720000000000000
dtype: object
```
This behaviour seems to stem from numpy's conversions as I mentioned in another comment.
I am not sure whether or not this is intentional but, to me, it seems a bit odd and I feel like the user would not be expecting to see such a divergence.